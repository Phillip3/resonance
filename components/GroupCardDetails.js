// @flow

import React from 'react';
import { StyleSheet, View, TouchableHighlight } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import * as Amplitude from 'expo-analytics-amplitude';
import Text from '../components/shared/Text';
import Colors from '../constants/Colors';
import {
  getMeetingFrequency,
  getMeetingDay,
  getMeetingTime,
} from '../utils/groups';

export default ({
  item,
}: {
  item: {
    name: string,
    campus: string,
    frequency: number,
    interval: string,
    daysOfWeek: Array<string>,
    dayOfMonth: string,
    meetingTime: string,
    description: string,
  },
}) => {
  const navigation = useNavigation();
  const {
    name = '',
    campus,
    frequency,
    interval,
    daysOfWeek,
    dayOfMonth,
    meetingTime,
    description,
  } = item;
  const titleParts = name.split('-');

  if (titleParts.length > 1) {
    titleParts.shift();
  }

  const title = titleParts.join(' ').trim();

  return (
    <TouchableHighlight
      style={styles.cardHighlight}
      underlayColor={Colors.darkestGray}
      onPress={() => {
        Amplitude.logEventWithProperties('OPEN Group Details', {
          group: title,
        });
        navigation.navigate('GroupDetails', {
          group: { ...item, title },
        });
      }}
    >
      <View style={styles.group}>
        <Text light adjustsFontSizeToFit numberOfLines={2} style={styles.title}>
          {title}
        </Text>

        <View style={styles.when}>
          <Text style={styles.detail}>
            {getMeetingFrequency(frequency, interval)} on{' '}
            <Text bold style={[styles.detail, styles.b]}>
              {getMeetingDay(daysOfWeek, dayOfMonth)}
            </Text>{' '}
            at{' '}
            <Text bold style={[styles.detail, styles.b]}>
              {getMeetingTime(meetingTime)}
            </Text>
          </Text>
        </View>

        <View style={styles.details}>
          <Text bold style={styles.detail}>
            {campus}
          </Text>
        </View>

        <View style={styles.descriptionWrapper}>
          <Text numberOfLines={4} style={styles.description}>
            {description}
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  );
};

export const styles = StyleSheet.create({
  cardHighlight: { borderRadius: 16 },
  group: {
    height: 250,
    padding: 16,
    borderRadius: 10,
    position: 'relative',
    overflow: 'hidden',
  },
  title: {
    marginTop: 16,
    marginBottom: 10,
    fontSize: 28,
    lineHeight: 30,
    color: Colors.white,
  },
  details: {
    marginBottom: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  detail: {
    fontSize: 16,
    color: Colors.gray,
  },
  b: { fontSize: 17 },
  when: {
    marginBottom: 10,
    justifyContent: 'center',
  },
  descriptionWrapper: {
    flex: 1,
    justifyContent: 'center',
  },
  description: {
    fontSize: 16,
    lineHeight: 24,
    color: Colors.gray,
  },
});
