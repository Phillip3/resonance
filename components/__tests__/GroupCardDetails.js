import React from 'react';
import { TouchableHighlight } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';
import { useNavigation } from '@react-navigation/native';
import GroupCardDetails from '../GroupCardDetails';

// mock react-navigation
jest.mock('@react-navigation/native');

describe('<GroupCardDetails/>', () => {
  test('render the group details', async () => {
    const mockNavigate = jest.fn();

    // mock the `useNavigation` hook
    useNavigation.mockReturnValue({ navigate: mockNavigate });

    const props = {
      item: {
        name: '01 - Test Group',
        campus: 'North San Jose',
        frequency: 1,
        interval: 'Weekly',
        daysOfWeek: ['Tuesday'],
        dayOfMonth: '',
        meetingTime: '19:00:00 PST',
        description: 'This group is a test',
      },
    };
    const { toJSON, getByType } = render(<GroupCardDetails {...props} />);

    expect(toJSON()).toMatchSnapshot();

    fireEvent.press(getByType(TouchableHighlight));

    expect(mockNavigate).toBeCalledWith('GroupDetails', {
      group: { ...props.item, title: 'Test Group' },
    });
  });
});
