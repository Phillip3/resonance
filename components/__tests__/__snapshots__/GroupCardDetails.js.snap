// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupCardDetails/> render the group details 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 16,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 10,
        "height": 250,
        "overflow": "hidden",
        "padding": 16,
        "position": "relative",
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      light={true}
      numberOfLines={2}
      selectable={true}
      style={
        Array [
          Object {
            "color": "#cbd2d6",
            "fontFamily": "NunitoSans-Light",
            "fontSize": 18,
            "lineHeight": 24,
            "textAlign": "auto",
          },
          Array [
            Object {
              "color": "#fff",
              "fontSize": 28,
              "lineHeight": 34,
              "marginBottom": 20,
            },
            Object {
              "marginBottom": 10,
              "marginTop": 16,
            },
          ],
        ]
      }
    >
      Test Group
    </Text>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        selectable={true}
        style={
          Array [
            Object {
              "color": "#cbd2d6",
              "fontFamily": "NunitoSans-Regular",
              "fontSize": 18,
              "lineHeight": 24,
              "textAlign": "auto",
            },
            undefined,
          ]
        }
      >
        Every week
         on
         
        <Text
          bold={true}
          selectable={true}
          style={
            Array [
              Object {
                "color": "#cbd2d6",
                "fontFamily": "NunitoSans-Bold",
                "fontSize": 18,
                "lineHeight": 24,
                "textAlign": "auto",
              },
              undefined,
            ]
          }
        >
          Tuesday
        </Text>
         at
         
        <Text
          bold={true}
          selectable={true}
          style={
            Array [
              Object {
                "color": "#cbd2d6",
                "fontFamily": "NunitoSans-Bold",
                "fontSize": 18,
                "lineHeight": 24,
                "textAlign": "auto",
              },
              undefined,
            ]
          }
        >
          7:00pm
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        bold={true}
        selectable={true}
        style={
          Array [
            Object {
              "color": "#cbd2d6",
              "fontFamily": "NunitoSans-Bold",
              "fontSize": 18,
              "lineHeight": 24,
              "textAlign": "auto",
            },
            undefined,
          ]
        }
      >
        North San Jose
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        light={true}
        numberOfLines={4}
        selectable={true}
        style={
          Array [
            Object {
              "color": "#cbd2d6",
              "fontFamily": "NunitoSans-Light",
              "fontSize": 18,
              "lineHeight": 24,
              "textAlign": "auto",
            },
            undefined,
          ]
        }
      >
        This group is a test
      </Text>
    </View>
  </View>
</View>
`;
